### PR TITLE
[FLINK-30216] Improve code quality by standardising tab/spaces in pom…

### PR DESCRIPTION
**Pull request by darenwkt**
## What is the purpose of the change

This change is to improve the code quality of the repo by standardising tab/spaces based on [flink code style guide](https://flink.apache.org/contributing/code-style-and-quality-formatting.html#whitespaces), which indicates that we should use spaces. The current repo has a mix of tab and spaces.

## Brief change log

- Added maven plugin for xml files checkstyle: https://www.mojohaus.org/xml-maven-plugin/check-format-mojo.html

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (no)
  - Core observer or reconciler logic that is regularly executed: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by gyfora — APPROVED** [review]
looks good! @mbalassi any objections?

File: pom.xml [hunk 1]
```diff
@@ -349,6 +349,36 @@ under the License.
...
+                <artifactId>xml-maven-plugin</artifactId>
+                <version>1.0.2</version>
+                <configuration>
+                    <indentSize>4</indentSize>
+                    <useDefaultFormatFileSet>false</useDefaultFormatFileSet>
+                    <formatFileSets>
+                        <formatFileSet>
+                            <directory>.</directory>
+                            <excludes>
+                                <exclude>**</exclude>
```
Comment (mbalassi) [inline pom.xml:364]: This seems unnecessary to me, the include is sufficient.

**Review by mbalassi — COMMENTED** [review]
Thanks @darenwkt. Could you please clarify my above comment? I am happy to merge otherwise.

File: pom.xml (hunk 1 above)
Comment (darenwkt) [reply pom.xml]: Hi @mbalassi, thank you for the feedback, I agree with your commend that include alone is sufficient. Have updated the PR, thank you

Comment (mbalassi) [reply pom.xml]: Roger that. I will merge as soon as the first build returns green.

**Review by mbalassi — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (darenwkt) [reply pom.xml]: Thank you!

**Pull request merged**